src/components/tl/mlx5/alltoall/alltoall_coll.c

```diff
@@ -0,0 +1,888 @@
+        if (status < 0) {
+            tl_error(UCC_TASK_LIB(task), "failed to start asr barrier");
+        }
+        for (i = 0; i < a2a->net.net_size; i++) {
+            task->alltoall.op->blocks_sent[i] = 0;
+            a2a->net.barrier
+                .flags[(a2a->net.net_size + 1) * task->alltoall.seq_index + i] =
+                task->alltoall.seq_num;
+        }
+        ucc_progress_enqueue(UCC_TL_CORE_CTX(team)->pq, coll_task);
+    } else {
+        tl_mlx5_barrier_t *local = tl_mlx5_barrier_local_addr(task);
+        *local                   = task->alltoall.seq_num;
+        for (i = 0; i < a2a->net.net_size; i++) {
+            task->alltoall.op->blocks_sent[i] = 0;
+            if (i == a2a->net.sbgp->group_rank) {
+                tl_mlx5_barrier_flag_set(task, i);
+                continue;
+            }
+
+            send_start(team, i);
+            status = send_block_data(
+                a2a, i, (uintptr_t)local, sizeof(tl_mlx5_barrier_t),
+                a2a->net.barrier.mr->lkey,
+                tl_mlx5_barrier_my_remote_addr(task, i),
+                tl_mlx5_barrier_remote_rkey(task, i), 0, NULL);
+            if (UCC_OK == status) {
+                status = send_done(team, i);
+            }
+            if (status != UCC_OK) {
+                tl_error(UCC_TASK_LIB(task), "failed  sending barrier notice");
+                return status;
+            }
+        }
+        coll_task->status = UCC_OK;
+        UCC_TL_MLX5_PROFILE_REQUEST_EVENT(task, "mlx5_alltoall_barreir_done",
+                                          0);
+        return ucc_task_complete(coll_task);
+    }
+    return UCC_OK;
+}
+
+static void ucc_tl_mlx5_asr_barrier_progress(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_mlx5_schedule_t *task = TASK_SCHEDULE(coll_task);
+    ucc_status_t            status;
+
+    status = ucc_collective_test(&task->alltoall.barrier_req->task->super);
+    if (status < 0) {
+        tl_error(UCC_TASK_LIB(task), "failure during asr barrier");
+    } else if (UCC_OK == status) {
+        tl_debug(UCC_TASK_LIB(task), "asr barrier done");
+        UCC_TL_MLX5_PROFILE_REQUEST_EVENT(task, "mlx5_alltoall_barreir_done",
+                                          0);
+        ucc_service_coll_finalize(task->alltoall.barrier_req);
+        coll_task->status = UCC_OK;
+    }
+}
+
+// add polling mechanism for blocks in order to maintain const qp tx rx
+static ucc_status_t ucc_tl_mlx5_send_blocks_start(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_mlx5_schedule_t *task           = TASK_SCHEDULE(coll_task);
+    ucc_tl_mlx5_team_t     *team           = SCHEDULE_TEAM(task);
+    ucc_tl_mlx5_alltoall_t *a2a            = team->a2a;
+    int                     node_size      = a2a->node.sbgp->group_size;
+    int                     net_size       = a2a->net.sbgp->group_size;
+    int                     op_msgsize     =
+                       node_size * a2a->max_msg_size * UCC_TL_TEAM_SIZE(team)
```

Comment:
alignment here and in all the next params as well until dm_host (including)...

Reply:
I did what I think is the convention: aligning the variable names and all the "=". clang-tidy does kind of a mess here. What do you want here exactly?

Reply:
Is it ok for the function `ucc_tl_mlx5_send_blocks_leftovers_start`?

Reply:
in my opinion it should either be keeping all params in the same alignment and then it would be like this:
    int                     op_msgsize     =
                                             node_size * a2a->max_msg_size *
                                             UCC_TL_TEAM_SIZE(team) *
                                             a2a->max_num_of_columns;

or make all the long ones on a seperate alignment like this:
    int op_msgsize =  node_size * a2a->max_msg_size * UCC_TL_TEAM_SIZE(team) *
                                    a2a->max_num_of_columns;

same should change for function ucc_tl_mlx5_send_blocks_leftovers_start...